File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if (s.length() == 0 || s.length() == 1)
			return s.length();
		int res = 0;
		vector<bool>vis(256, false);
		int left = 0, right = 0;
		while (right < s.length()) {
			while (vis[s[right]]) {
				
				vis[s[left]] = false;
				left++;
			}
			vis[s[right]] = true;
			res = max(res, (right - left + 1));
			right++;
		}
		return res;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if (len(s)==0 or len(s)==1):
            return len(s)
        left=0
        right=0
        vis=[False]*256
        res=0
        while right<len(s):
            while vis[ord(s[right])]==True:
                vis[ord(s[left])]=False
                left+=1
            vis[ord(s[right])]=True
            res=max(res,right-left+1)
            right+=1
        return res
        